java/parquet/src/main/java/sleeper/io/parquet/record/ParquetRecordWriterFactory.java

```diff
@@ -0,0 +1,116 @@
+ * limitations under the License.
+ */
+package sleeper.io.parquet.record;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.api.WriteSupport;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.schema.MessageType;
+
+import sleeper.configuration.properties.InstanceProperties;
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.core.record.Record;
+import sleeper.core.schema.Schema;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+import static sleeper.configuration.properties.table.TableProperty.COMPRESSION_CODEC;
+import static sleeper.configuration.properties.table.TableProperty.DICTIONARY_ENCODING_FOR_ROW_KEY_FIELDS;
+import static sleeper.configuration.properties.table.TableProperty.DICTIONARY_ENCODING_FOR_SORT_KEY_FIELDS;
+import static sleeper.configuration.properties.table.TableProperty.DICTIONARY_ENCODING_FOR_VALUE_FIELDS;
+import static sleeper.configuration.properties.table.TableProperty.PAGE_SIZE;
+import static sleeper.configuration.properties.table.TableProperty.ROW_GROUP_SIZE;
+
+public class ParquetRecordWriterFactory {
+
+    private ParquetRecordWriterFactory() {
+    }
+
+    public static ParquetWriter<Record> createParquetRecordWriter(Path path, Schema schema) throws IOException {
+        return new Builder(path, SchemaConverter.getSchema(schema), schema).build();
+    }
+
+    public static ParquetWriter<Record> createParquetRecordWriter(Path path, TableProperties tableProperties, Configuration conf) throws IOException {
+        return createParquetRecordWriter(path,
+            tableProperties.getSchema(),
+            tableProperties.get(COMPRESSION_CODEC),
+            tableProperties.getLong(ROW_GROUP_SIZE),
+            tableProperties.getInt(PAGE_SIZE),
+            tableProperties.getBoolean(DICTIONARY_ENCODING_FOR_ROW_KEY_FIELDS),
+            tableProperties.getBoolean(DICTIONARY_ENCODING_FOR_SORT_KEY_FIELDS),
+            tableProperties.getBoolean(DICTIONARY_ENCODING_FOR_VALUE_FIELDS),
+            conf);
+    }
+
+    public static ParquetWriter<Record> createParquetRecordWriter(Path path, Schema schema, Configuration conf) throws IOException {
+        TableProperties tableProperties = new TableProperties(new InstanceProperties());
+        tableProperties.setSchema(schema);
+        return createParquetRecordWriter(path, tableProperties, conf);
+    }
+
+    public static ParquetWriter<Record> createParquetRecordWriter(Path path,
+            Schema schema,
+            String compressionCodec,
+            long rowGroupSize,
+            int pageSize,
+            boolean dictionaryEncodingForRowKeyFields,
+            boolean dictionaryEncodingForSortKeyFields,
+            boolean dictionaryEncodingForValueFields,
+            Configuration conf) throws IOException {
```

Comment:
This method could return a builder rather than take all the parameters at once.

We could move the dictionary encoding flags to the builder with methods named withDictionaryEncodingForXFields, especially since we're passing in the schema in the constructor.

We could add an overload for withCompressionCodec to take a string if that seems worth doing.

Reply:
Last 2 points have been actioned. The first point will be considered in the S3ResultsOutput issue.